lib/jnpr/junos/utils/start_shell.py

```diff
@@ -21,28 +22,39 @@ def _ssh_exec(self, command):
 
     """
 
-    def __init__(self, nc):
+    def __init__(self, nc, timeout=30):
         """
         Utility Constructor
 
         :param Device nc: The Device object
+
+        :param int timeout:
+          Timeout value in seconds to wait for expected string/pattern.
         """
         self._nc = nc
+        self.timeout = timeout
 
-    def wait_for(self, this=_SHELL_PROMPT):
+    def wait_for(self, this=_SHELL_PROMPT, timeout=0):
```

Comment:
We also need to add a `timeout=0` argument to the `run()` method and pass that `timeout` value through to the `wait_for()` method. Otherwise, you can't specify a unique timeout for a specific command. You just get the timeout specified when the `StartShell` object was created.